ecs-cli/modules/commands/flags/flags.go

```diff
@@ -182,6 +183,14 @@ func OptionalCreateLogsFlag() cli.Flag {
 	}
 }
 
+// OptionalCreateLogsFlag allows users to force an update of running tasks or services
+func OptionalForceUpdateFlag() cli.Flag {
```

Comment:
Isn't in redundant to have this on Services, since `--force-deployment` is already a valid flag? What does this do that the other flag does not?

Reply:
I'm pretty sure she only is using it for `compose up`, but I agree that the help text should change.

Reply:
It's not on services -- this flag only is on the `upCommand` in `compose_command.go`. See below for valid flags on `compose up` vs `compose service up` -- but I see that the comment is inaccurate so I will fix that!
```
~/go/src/github.com/aws/amazon-ecs-cli(task-placement)$ ecs-cli compose up --help
NAME:
   ecs-cli compose up - Creates an ECS task definition from your compose file (if it does not already exist) and runs one instance of that task on your cluster (a combination of create and start).

USAGE:
   ecs-cli compose up [command options] [arguments...]

OPTIONS:
   --region value, -r value   [Optional] Specifies the AWS region to use. Defaults to the region configured using the configure command
   --cluster-config value     [Optional] Specifies the name of the ECS cluster configuration to use. Defaults to the default cluster configuration.
   --ecs-profile value        [Optional] Specifies the name of the ECS profile configuration to use. Defaults to the default profile configuration. [$ECS_PROFILE]
   --aws-profile value        [Optional] Use the AWS credentials from an existing named profile in ~/.aws/credentials. [$AWS_PROFILE]
   --cluster value, -c value  [Optional] Specifies the ECS cluster name to use. Defaults to the cluster configured using the configure command
   --launch-type value        [Optional] Specifies the launch type. Options: EC2 or FARGATE. Overrides the default launch type stored in your cluster configuration. Defaults to EC2 if a cluster configuration is not used.
   --create-log-groups        [Optional] Create the CloudWatch log groups specified in your compose file(s).
   --force-update, -u         [Optional] Forces update of task or service with current run parameters
```

```
~/go/src/github.com/aws/amazon-ecs-cli(task-placement)$ ecs-cli compose service up --help
NAME:
   ecs-cli compose service up - Creates a new ECS service or updates an existing one according to your compose file. For new services or existing services with a current desired count of 0, the desired count for the service is set to 1. For existing services with non-zero desired counts, a new task definition is created to reflect any changes to the compose file and the service is updated to use that task definition. In this case, the desired count does not change.

USAGE:
   ecs-cli compose service up [command options] [arguments...]

OPTIONS:
   --deployment-max-percent value          [Optional] Specifies the upper limit (as a percentage of the service's desiredCount) of the number of running tasks that can be running in a service during a deployment. Defaults to 200.
   --deployment-min-healthy-percent value  [Optional] Specifies the lower limit (as a percentage of the service's desiredCount) of the number of running tasks that must remain running and healthy in a service during a deployment. Defaults to 100.
   --target-group-arn value                [Optional] Specifies the full Amazon Resource Name (ARN) of a previously configured Elastic Load Balancing target group to associate with your service.
   --container-name value                  [Optional] Specifies the container name (as it appears in a container definition). This parameter is required if a load balancer or target group is specified.
   --container-port value                  [Optional] Specifies the port on the container to associate with the load balancer. This port must correspond to a containerPort in the service's task definition. This parameter is required if a load balancer or target group is specified.
   --load-balancer-name value              [Optional] Specifies the name of a previously configured Elastic Load Balancing load balancer to associate with your service.
   --role value                            [Optional] Specifies the name or full Amazon Resource Name (ARN) of the IAM role that allows Amazon ECS to make calls to your load balancer or target group on your behalf. This parameter is required if you are using a load balancer or target group with your service. If you specify the role parameter, you must also specify a load balancer name or target group ARN, along with a container name and container port.
   --health-check-grace-period value       [Optional] Specifies the period of time, in seconds, that the Amazon ECS service scheduler should ignore unhealthy Elastic Load Balancing target health checks after a task has first started.
   --region value, -r value                [Optional] Specifies the AWS region to use. Defaults to the region configured using the configure command
   --cluster-config value                  [Optional] Specifies the name of the ECS cluster configuration to use. Defaults to the default cluster configuration.
   --ecs-profile value                     [Optional] Specifies the name of the ECS profile configuration to use. Defaults to the default profile configuration. [$ECS_PROFILE]
   --aws-profile value                     [Optional] Use the AWS credentials from an existing named profile in ~/.aws/credentials. [$AWS_PROFILE]
   --cluster value, -c value               [Optional] Specifies the ECS cluster name to use. Defaults to the cluster configured using the configure command
   --timeout value                         Specifies the timeout value in minutes (decimals supported) to wait for the running task count to change. If the running task count has not changed for the specified period of time, then the CLI times out and returns an error. Setting the timeout to 0 will cause the command to return without checking for success. (default: 5)
   --launch-type value                     [Optional] Specifies the launch type. Options: EC2 or FARGATE. Overrides the default launch type stored in your cluster configuration. Defaults to EC2 if a cluster configuration is not used.
   --create-log-groups                     [Optional] Create the CloudWatch log groups specified in your compose file(s).
   --force-deployment                      [Optional] Whether or not to force a new deployment of the service.
```

Reply:
Makes sense! And yes, I was totally going by what the comment said it did vs. looking at the implementation :P